backends/sycl-ref/ceed-sycl-vector.sycl.cpp

```diff
@@ -466,10 +466,46 @@ static int CeedVectorGetArrayWrite_Sycl(const CeedVector vec, const CeedMemType
 // Get the norm of a CeedVector
 //------------------------------------------------------------------------------
 static int CeedVectorNorm_Sycl(CeedVector vec, CeedNormType type, CeedScalar *norm) {
-  // TODO
   Ceed ceed;
   CeedCallBackend(CeedVectorGetCeed(vec, &ceed));
-  return CeedError(ceed, CEED_ERROR_BACKEND, "Ceed SYCL function not implemented");
+  CeedVector_Sycl *impl;
+  CeedCallBackend(CeedVectorGetData(vec, &impl));
+  CeedSize length;
+  CeedCallBackend(CeedVectorGetLength(vec, &length));
+  Ceed_Sycl *data;
+  CeedCallBackend(CeedGetData(ceed, &data));
+
+  // Compute norm
+  const CeedScalar *d_array;
+  CeedScalar       *reduced_value = sycl::malloc_device<CeedScalar>(1, data->sycl_device, data->sycl_context);
```

Comment:
Let's move this variable to be a member of `CeedVector_Sycl` so that it can be allocated/deallocated once.

Reply:
`reduced_value` has been moved to be a member of `CeedVector_Sycl` class. Variable renamed as `reduction_norm` for better clarity.